evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -142,4 +142,13 @@ def validate(self, X, y):
                                                      message_code=DataCheckMessageCode.TARGET_INCOMPATIBLE_OBJECTIVE,
                                                      details=details).to_dict())
 
+        if X is not None:
+            X = _convert_to_woodwork_structure(X)
+            X_index = list(X.to_dataframe().index)
+            y_index = list(y_df.index)
+            if X_index != y_index:
+                messages["warnings"].append(DataCheckWarning(message="Input target and features have mismatched indices",
+                                                             data_check_name=self.name,
+                                                             message_code=DataCheckMessageCode.MISMATCHED_INDICES,
+                                                             details={"feature_index": X_index, "target_index": y_index}).to_dict())
```

Comment:
I don't think it's a good idea to return the entire indices of X and y because they can be really big. Maybe we just return a sample of the diff?

Reply:
@freddyaboulton Ooo yes, I love this suggestion!! Do you think returning the entire diff is okay, or should we do something like use a parameter `n_unique` to determine how many we should display / keep like we have for unique target values?

Pro: more flexibility
Con: another parameter

Or I could also combine it with n_unique / make that handle both cases 🤔